### DISCUSSION TITLE
Grain size distribution

### DISCUSSION THREAD

**Discussion by nitheeshkumar11** [neperfepx/neper#739] [Q&A] [answered]
Greetings, 
I want to generate a microstructure with grain size distribution of 49+-12 micrometers in a 2mm cubic domain. How do I do that?

Thanks in advance.

**Comment by rquey** [comment, marked as answer]
You would typically specify a lognormal grain size distribution of average 0.049 and standard deviation of your choice.  It is also a good idea to specify a sphericity distribution:
```
neper -T -n from_morpho -morpho "diameq:lognormal(0.049,0.015),1-sphericity:lognormal(0.145,0.03)" -domain "cube(0.2,0.2,0.2)"
```

To exclude values smaller than 49 - 12 and larger than 49 + 12 micrometers, add `from` and `to`:
```
neper -T -n from_morpho -morpho "diameq:lognormal(0.049,0.015,from=0.037,to=0.061),1-sphericity:lognormal(0.145,0.03)" -domain "cube(0.2,0.2,0.2)"
neper -V test.tess -print img
```
![d](https://github.com/neperfepx/neper/assets/2659376/4825e5a4-1152-4845-baa3-1a612ba08cdc)

Note that I used a domain of size 0.2 (mm), since 2 mm leads to 100,000+ cells.
